### DISCUSSION TITLE
Option for `np.log1p` in Log Transformer

### DISCUSSION THREAD

**Discussion by ilkersigirci** [alan-turing-institute/sktime#3275] [Ideas]
For small values that are close or equal to 0, currently Log transformer gives error since it uses `np.log(data)` in transformer and `exp(data)` for inverse transformer. It would be [better](https://stackoverflow.com/a/49538384/17024775) if we can use `np.log1p(data)` and `expm1(data)` as another option. This can be easily done with one more init parameter of the Log transformer. What did you think about this?

**Comment by fkiraly** [comment]
Why not - or, more generally, an `offset` parameter?

**Reply by ilkersigirci** [reply]
You are right, the `offset` approach is more generic imo, nice thought

**Reply by fkiraly** [reply]
opened issue here: https://github.com/alan-turing-institute/sktime/issues/3286

**Comment by ilkersigirci** [comment]
Implemented in #3354. Thanks.
